### PR TITLE
v-tooltip retrofit

### PR DESCRIPTION
**Pull request by stuartromanek**
Implements v-tooltip (https://github.com/Akryum/v-tooltip) and replaces our hand rolled positioning of context menus. https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4?selectedIssue=A30U-447

We will also use this library for real tooltips. 

Most of this PR is retrofitting and making sure all our existing context menu implementations and their events work properly in the new setup

### CODE REVIEW

File: modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue [hunk 1]
```diff
@@ -7,7 +7,7 @@
       <AposEmptyState :empty-state="emptyState" />
       <AposAreaMenu
         @add="insert"
-        :context-options="contextOptions"
+        :context-menu-options="contextMenuOptions"
```
Comment (stuartromanek) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue:10]: renamed for clarity

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue [hunk 2]
```diff
@@ -21,20 +21,19 @@
...
               class="apos-admin-bar__create"
-              :menu="createMenu" :button="{
+              :menu="createMenu"
+              :button="{
                 label: 'New item',
                 iconOnly: true,
                 icon: 'plus-icon',
                 type: 'primary'
               }"
-              tip-alignment="right"
```
Comment (stuartromanek) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue:37]: Context menu UI no longer uses `tip-alignment` or `position`

File: modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue [hunk 3]
```diff
@@ -97,8 +97,7 @@ export default {
       editing: {},
       hoveredWidget: null,
       focusedWidget: null,
-      contextOptions: {
-        autoPosition: false,
```
Comment (stuartromanek) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue:101]: no more `autoPosition`

File: modules/@apostrophecms/modal/ui/apos/components/AposModal.vue [hunk 4]
```diff
@@ -126,13 +126,13 @@ export default {
     },
     gridModifier() {
       if (this.hasLeftRail && this.hasRightRail) {
-        return 'apos-modal__main--grid apos-modal__main--with-rails';
+        return 'apos-modal__main--with-rails';
```
Comment (stuartromanek) [inline modules/@apostrophecms/modal/ui/apos/components/AposModal.vue:129]: `.apos-modal__main` is now display: grid all the time. It children being able to fill empty modal space lets v-tooltip get positioned correctly. Not a great reason for this change but =/

File: modules/@apostrophecms/ui/ui/apos/components/AposContextMenu.stories.js [hunk 5]
```diff
@@ -1,89 +1,73 @@
-import { storiesOf } from '@storybook/vue';
-
 import AposContextMenu from './AposContextMenu.vue';
+import {
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/components/AposContextMenu.stories.js:2]: update context menu story format

File: modules/@apostrophecms/ui/ui/apos/components/AposContextMenuTip.vue [hunk 6]
```diff
@@ -22,29 +22,22 @@ export default {
...
       return `apos-context-menu__tip--alignment-${this.align}`;
-    },
-    originModifier () {
-      return `apos-context-menu__tip--origin-${this.origin}`;
     }
   }
 };
 </script>
 
-<style lang="scss" scoped>
```
Comment (stuartromanek) [inline modules/@apostrophecms/ui/ui/apos/components/AposContextMenuTip.vue:42]: context menus now get yanked to the top of the DOM, making scoping a no-go

**Review by stuartromanek — COMMENTED** [review]
Comments

**Review by boutell — APPROVED** [review]

**Review by abea — APPROVED** [review]

**Pull request merged**